server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetSettingsAction.java

```diff
@@ -73,31 +67,16 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
                 .indices(Strings.splitStringByCommaToArray(request.param("index")))
                 .indicesOptions(IndicesOptions.fromRequest(request, IndicesOptions.strictExpandOpen()))
                 .humanReadable(request.hasParam("human"))
+                .includeDefaults(renderDefaults)
                 .names(names);
         getSettingsRequest.local(request.paramAsBoolean("local", getSettingsRequest.local()));
+        getSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", getSettingsRequest.masterNodeTimeout()));
 
         return channel -> client.admin().indices().getSettings(getSettingsRequest, new RestBuilderListener<GetSettingsResponse>(channel) {
 
             @Override
             public RestResponse buildResponse(GetSettingsResponse getSettingsResponse, XContentBuilder builder) throws Exception {
-                builder.startObject();
-                for (ObjectObjectCursor<String, Settings> cursor : getSettingsResponse.getIndexToSettings()) {
-                    // no settings, jump over it to shorten the response data
-                    if (cursor.value.isEmpty()) {
-                        continue;
-                    }
-                    builder.startObject(cursor.key);
-                    builder.startObject("settings");
-                    cursor.value.toXContent(builder, request);
-                    builder.endObject();
-                    if (renderDefaults) {
-                        builder.startObject("defaults");
-                        settingsFilter.filter(indexScopedSettings.diff(cursor.value, settings)).toXContent(builder, request);
-                        builder.endObject();
-                    }
-                    builder.endObject();
-                }
-                builder.endObject();
+                getSettingsResponse.toXContent(builder, request);
```

Comment:
one more thing, can we also adapt `RestGetAllSettingsAction` like this one?

Reply:
We decided off-line to do this in another piece of work as RestGetAllSettingsAction is structured differently